### ISSUE TITLE
ValueError after adding a Dropout layer

### ISSUE THREAD

**Issue by simonzack** [fchollet/keras#678]
I'm encountering the following error, but not sure why as the network looks quite reasonable

    Inputs types: [TensorType(float64, matrix), TensorType(float64, matrix)]
    Inputs shapes: [(50, 300), (50, 100)]
    Inputs strides: [(2400, 8), (800, 8)]
    Inputs values: ['not shown', 'not shown']

This is a simplified version of my current network which reproduces the error:

    import numpy as np
    from keras.layers.core import Dense, Dropout, Flatten
    from keras.layers.embeddings import Embedding
    from keras.models import Graph

    model = Graph()
    model.add_input('input', dtype='int32')
    model.add_node(Embedding(3, 10), 'embedding', 'input')
    model.add_node(Flatten(), 'flatten_0', 'embedding')
    model.add_node(Flatten(), 'flatten_1', 'embedding')
    model.add_node(Flatten(), 'flatten_2', 'embedding')
    model.add_node(Dropout(0.5), 'dropout', inputs=['flatten_0', 'flatten_1', 'flatten_2'])
    model.add_node(Dense(90, 3, activation='softmax'), 'output', 'dropout', create_output=True)
    model.compile('adadelta', {'output': 'mse'})

    X = np.array([[0, 0, 0], [1, 1, 1], [2, 2, 2]])
    y = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
    model.fit({'input': X, 'output': y}, batch_size=1, nb_epoch=8)

Everything works however without the Dropout layer:

    import numpy as np
    from keras.layers.core import Dense, Dropout, Flatten
    from keras.layers.embeddings import Embedding
    from keras.models import Graph

    model = Graph()
    model.add_input('input', dtype='int32')
    model.add_node(Embedding(3, 10), 'embedding', 'input')
    model.add_node(Flatten(), 'flatten_0', 'embedding')
    model.add_node(Flatten(), 'flatten_1', 'embedding')
    model.add_node(Flatten(), 'flatten_2', 'embedding')
    model.add_node(Dense(90, 3, activation='softmax'), 'output', inputs=['flatten_0', 'flatten_1', 'flatten_2'], create_output=True)
    model.compile('adadelta', {'output': 'mse'})

    X = np.array([[0, 0, 0], [1, 1, 1], [2, 2, 2]])
    y = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
    model.fit({'input': X, 'output': y}, batch_size=1, nb_epoch=8)

**Comment by simonzack** [comment]
There seems to be some problem with `X.shape`

In `Dropout`, if I change X.shape to the right one it works

    X *= self.srng.binomial((1, 90), p=retain_prob, dtype=theano.config.floatX)

**Comment by simonzack** [comment]
Apparently a theano bug, installing theano 0.7.1a1 fixed this for me.
